### ISSUE TITLE
Increase ByteBuf writer index when HttpData is converted to ByteBuf

### ISSUE THREAD

**Issue by khrupalyk** [zio/zio-http#1220]
When HttpData of BinaryStream type is converted to ByteBuf addComponent is used to accumulate data in the BB

By default, addComponent doesn't update writerIndex

```
public CompositeByteBuf addComponent(ByteBuf buffer) {
    return addComponent(increaseWriterIndex = false, buffer);
 }
```
So to get data from BB we need first update writerIndex
```
val stream = ZStream.fromIterable(1 to 10000 map (_.toString))

HttpData.fromStream(stream).toByteBuf.map { bb =>
  println(bb.readableBytes()) // 0
  println(bb.toString(Charset.defaultCharset()).length) //0

  // update write index
  bb.writerIndex(bb.capacity())

  println(bb.readableBytes()) //38894
  println(bb.toString(Charset.defaultCharset()).length) //38894
}.as(ExitCode.success).orDie
```

Is there a reason why this is not done?

How it implemented now
```
  def toByteBuf: Task[ByteBuf] = {
    self match {
      case HttpData.Text(text, charset)   => UIO(Unpooled.copiedBuffer(text, charset))
      case HttpData.BinaryChunk(data)     => UIO(Unpooled.copiedBuffer(data.toArray))
      case HttpData.BinaryByteBuf(data)   => UIO(data)
      case HttpData.Empty                 => UIO(Unpooled.EMPTY_BUFFER)
      case HttpData.BinaryStream(stream)  =>
        stream
          .asInstanceOf[ZStream[Any, Throwable, ByteBuf]]
          .fold(Unpooled.compositeBuffer())((c, b) => c.addComponent(b))
      case HttpData.RandomAccessFile(raf) =>
        effectBlocking {
          val fis                      = new FileInputStream(raf().getFD)
          val fileContent: Array[Byte] = new Array[Byte](raf().length().toInt)
          fis.read(fileContent)
          Unpooled.copiedBuffer(fileContent)
        }
    }
  }
```

**Comment by amitksingh1490** [comment]
Thanks for reporting this. You are right we need to update the writer Index. This is a bug. BTW which version are you using?

**Comment by khrupalyk** [comment]
I'm using 1.0.0.0-RC25, but I've checked that it's already fixed for 1.0.0.0-RC27
This issue could be closed.
Sorry for disturbing.
